test/org/sagebionetworks/bridge/services/ParticipantServiceTest.java

```diff
@@ -626,6 +636,92 @@ public void notSettingStatusDoesntClearStatus() {
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account, never()).setEmail("updated@email.com");
+        verify(account, never()).setEmailVerified(Boolean.FALSE);
+        verify(accountWorkflowService, never()).sendEmailVerificationToken(STUDY, ID, "updated@email.com");
+    }
+
+    @Test
+    public void updateParticipantEmailCanBeAddedWithVerification() {
+        STUDY.setEmailVerificationEnabled(true);
+        mockHealthCodeAndAccountRetrieval(null, PHONE);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withEmail("updated@email.com").withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account).setEmail("updated@email.com");
+        verify(account).setEmailVerified(Boolean.FALSE);
+        verify(accountWorkflowService).sendEmailVerificationToken(STUDY, ID, "updated@email.com");
+    }
+
+    @Test
+    public void updateParticipantEmailCanBeAddedNoVerification() {
+        STUDY.setEmailVerificationEnabled(false);
+        mockHealthCodeAndAccountRetrieval(null, PHONE);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withEmail("updated@email.com").withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account).setEmail("updated@email.com");
+        verify(account).setEmailVerified(Boolean.FALSE);
+        verify(accountWorkflowService, never()).sendEmailVerificationToken(STUDY, ID, "updated@email.com");
+    }
+    
+    @Test
+    public void updateParticipantPhoneCannotBeChanged() {
+        mockHealthCodeAndAccountRetrieval(EMAIL, PHONE);
+        when(account.getPhoneVerified()).thenReturn(Boolean.TRUE);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().copyOf(PARTICIPANT)
+                .withPhone(OTHER_PHONE).withEmailVerified(Boolean.FALSE).build();
+        
+        participantService.updateParticipant(STUDY, ImmutableSet.of(), participant);
+        
+        verify(accountDao).updateAccount(accountCaptor.capture());
+        Account account = accountCaptor.getValue();
+        verify(account, never()).setPhone(PHONE);
+        verify(account, never()).setPhoneVerified(Boolean.TRUE);
+        verify(accountWorkflowService, never()).sendEmailVerificationToken(STUDY, ID, EMAIL);
```

Comment:
Similarly, is it worth changing that we never call sendEmailVerificationToken(any()), just to make sure we're not calling it with null? (Here and in the next test.)

Reply:
Yes, changed here as well (and in the other tests that are verifying the method was never called).